src/config_format/flb_cf_yaml.c

```diff
@@ -481,15 +481,23 @@ static int consume_event(struct flb_cf *cf, struct local_ctx *ctx,
                 return YAML_FAILURE;
             }
 
-            /* value is the 'custom plugin name', create a section instance */
-            if (flb_cf_section_property_add(cf, s->cf_section->properties,
-                                            "name", 4,
-                                            value, len) < 0) {
-                return YAML_FAILURE;
-            }
+            if (strcmp(value, "name") != 0) {
+                /* value is the 'custom plugin name', create a section instance */
+                if (flb_cf_section_property_add(cf, s->cf_section->properties,
+                                                "name", 4,
+                                                value, len) < 0) {
+                    return YAML_FAILURE;
+                }
 
-            /* next state are key value pairs for the custom plugin*/
-            s->state = STATE_CUSTOM_KEY_VALUE_PAIR;
+                /* next state are key value pairs for the custom plugin*/
+                s->state = STATE_CUSTOM_KEY_VALUE_PAIR;
+                flb_warn("using old 'name: *properties' format for custom section. "
+                         "this format is deprecated.");
+            } else {
+                s->state = STATE_CUSTOM_VAL;
+                value = (char *) event->data.scalar.value;
+                s->key = flb_sds_create(value);
```

Comment:
Verify the result of `flb_sds_create` and if possible amend the remaining occurrences of this issue at least in this function. 

Even if we assume that this will be called when the program starts which makes us think this shouldn't fail there are probably corner cases and not checking results is simply not a good habit.

As a side note, if a hot reload causes this to be called then it makes even more sense to verify it.